app/controllers/application_pets_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ApplicationPetsController < ApplicationController
```

Comment:
Great controller organization! Using a joins controller here and a `create` action is a very RESTful choice

You don't actually need the `new` action here, though, as that would be for a form for creating a joins record, and you are just triggering that creation from the applications show page